src/coreclr/src/vm/eventpipemetadatagenerator.cpp

```diff
@@ -16,7 +16,7 @@ BYTE* EventPipeMetadataGenerator::GenerateEventMetadata(
     EventPipeEventLevel level,
     EventPipeParameterDesc *pParams,
     unsigned int paramCount,
-    size_t &metadataLength)
+    size_t *metadataLength)
```

Comment:
Why do we need to make this parameter a pointer? What's wrong with reference?

Reply:
Usually the runtime prefers pointers - it's explicit at call site that the data can be modified. Not sure if there are any technical reasons beyond this.

Reply:
It doesn't appear to be documented in clr, but the runtime has always banned the use of "by reference" arguments in our coding conventions. It is not clear that the argument can change from the call site, like @hoyosjs mentions above.

Since we use pointers for all arguments that can be modified it leads to confusion when by reference variables are used. The first time I read through the EventPipe code it took me a minute to figure out how metadatalength was being changed since we don't use by reference.

I don't believe it is good practice to have different subsystems in the runtime follow drastically different coding conventions.